server/odc-core/src/main/java/com/oceanbase/odc/core/sql/split/SqlCommentProcessor.java

```diff
@@ -199,39 +213,49 @@ private synchronized void addLineMysql(List<String> sqls, StringBuffer buffer, S
             }
             // 扫描到转义字符，可能出现指令
             if ((!mlComment && inChar == '\\')) {
-                if ((inChar = lines[++pos]) == 0) {
+                inOrderChar = lines[++pos];
+                inChar = inOrderChar.getCh();
+                if (inChar == 0) {
                     break;
                 }
                 if (inString != '\0' || inChar == 'N') {
-                    lines[out++] = '\\';
+                    lines[out++] = lines[pos - 1];
                     if (inChar == '`' && inString == inChar) {
                         pos--;
                     } else {
-                        lines[out++] = inChar;
+                        lines[out++] = lines[pos];
```

Comment:
why should we change this?

Reply:
As lines[pos] is a reference of `OrderChar` rather than a primitive data type any more.